src/api-calls.js

```diff
@@ -9,13 +9,21 @@ export const getOembed = (desiredClip) => {
         };
       }
       const video_id = oembed.html.substring(oembed.html.search(/data-video-id="/g) + 15, oembed.html.search(/data-video-id="/g) + 34);
+      const startSound = oembed.html.search(/https:\/\/www.tiktok.com\/music/g);
+      const soundString = oembed.html.substring(startSound);
+      const endSound = soundString.search(/">/);
+      const endTitle = soundString.search(/<\/a>/);
+      const sound_url = oembed.html.slice(startSound, startSound + endSound);
+      const sound_title = soundString.slice(endSound + 2, endTitle);
```

Comment:
These are the RegWx statements that could use some refactoring, if you need to see a reference of what the response looks like just check the forIndependent.json integration in the cypress directory

Reply:
I think what you have here has the benefit of being pretty readable. (A common complaint I've heard about regex searches is that they're hard to read/interpret.) But you could probably streamline it a bit using parentheses to capture any values you want to retain. I think you could do this in combination with the `match` method that will return an array with your matches. So you could use one set of parens to capture the sound url and another set of parens to capture the sound title. That would let you avoid capturing only the index for those things and then using the indexes to slice. I'm not sure the end result would be more readable, but it might be!

I'm thinking it could look something like this:
`const myValues = oembed.html.match( [pattern for beginning of search ([pattern for matching url]) [pattern for moving to title area] ([pattern for matching title])`

and then you could just grab the two captured sections out of the resulting array and store them in `sound_url` and `sound_title`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match